Added base/mapping quality filters for HetPulldown

samuelklee opened the pull request:
Also removed hetAlleleFraction parameter as it was a holdover from previous versions of HetPulldown where allelic skew was given as input.

samuelklee commented:
Evaluation is not quite complete, but it looks like results slightly improve or otherwise do not change drastically.  In any case, it seems that some sort of filtering should be done in this step, so I am of the opinion that we should merge even if concordance with HAPSEG decreases.

This was prompted by #355 but does not completely resolve that issue.  Looks like two hets are consistently pulled down on Y across a few normals...will continue investigating.

LeeTL1220 commented:
Despite the fact that we still see hets on Y, I am assuming that I should still review this PR.

LeeTL1220 commented on src/main/java/org/broadinstitute/hellbender/tools/exome/GetHetCoverage.java:
```diff
@@ -72,14 +69,6 @@
     protected File tumorHetOutputFile;
 
     @Argument(
-            doc = "Heterozygous allele fraction.",
```
Does this mean that we need to change the workflow scripts?

samuelklee replied:
yup, should be a minor change.  will issue a PR for it.

LeeTL1220 commented on src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java:
```diff
@@ -31,8 +31,12 @@
+
     /** Set quality and read-depth thresholds for pulldown, interval threshold for indexing for SamLocusIterator. */
-    private static final int MIN_QUALITY = 0;
+    private static final int MIN_BASE_QUALITY = 20;
```
Are you sure that you want these hardcoded and not an advanced parameter with a default value?

samuelklee replied:
done, exposed as minMQ and minBQ

LeeTL1220 commented on src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java:
```diff
@@ -31,8 +31,12 @@
     private final File refFile;
     private final IntervalList snpIntervals;
 
+    private static final int NUMBER_OF_LOG_UPDATES = 20;
```
What is a log update?  Can you add a comment?

samuelklee replied:
done

LeeTL1220 commented:
Minor comments... Back to you @samuelklee

samuelklee commented:
Thanks, @LeeTL1220, I've addressed the PR comments.  Also, evaluation finished on the PRAD samples and it looks like MAE and RMSE either slightly improve or don't change much across various cuts of the samples.  So even though we lose ~5% of the old hets I'm comfortable with this change.
